my-project/src/Controller/RegisterController.php

```diff
@@ -0,0 +1,53 @@
+<?php
+/**
+ * VI-55 Registration
+ *
+ * @category   RegistrationController
+ * @package    Virtua_Registration
+ * @copyright  Copyright (c) Virtua
+ * @author     Mateusz Soboń <m.sobon@wearevirtua.com>
+ */
+
+namespace App\Controller;
+
+use App\Entity\User;
+use App\Form\UserType;
+use App\Repository\RoleRepository;
+use Symfony\Component\HttpFoundation\RedirectResponse;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\Routing\Annotation\Route;
+use Symfony\Bundle\FrameworkBundle\Controller\Controller;
+use Symfony\Component\Security\Core\Encoder\UserPasswordEncoderInterface;
+
+class RegisterController extends Controller
+{
+    /**
+     * @Route("/register", name="userRegister")
+     */
+    public function index(Request $request, UserPasswordEncoderInterface $passwordEncoder, RoleRepository $roleRepository)
+    {
+        $user = new User();
+        $form = $this->createForm(UserType::class, $user);
+        $form->handleRequest($request);
+
+        if($form->isSubmitted() && $form->isValid())
+        {
```

Comment:
@mSobonVirtua we should add additional valitation for form fields

Reply:
@astyczen What do you mean? I tested validation and it working fine.

Reply:
@mSobonVirtua like trimming, stripping slashes and other safety issues

Reply:
@astyczen I read about it and symfony do things like this automatically.